R/boxr_auth.R

```diff
@@ -431,32 +431,67 @@ box_auth_service <- function(token_file = NULL, token_text = NULL) {
   
   # build out a claim/payload as a specific user
   auth_url <- "https://api.box.com/oauth2/token"
-  claim <- jose::jwt_claim(
-    iss = config$boxAppSettings$clientID,
-    sub = as.character(user_id), # maybe don't need this? (can't hurt)
-    box_sub_type = "enterprise", # opinion - too risky to support user auth
-    aud = auth_url,
-    jti = openssl::base64_encode(openssl::rand_bytes(16)),
-    exp = unclass(Sys.time()) + 45
-  )
   
-  # sign claim with key
-  assertion <- jose::jwt_encode_sig(
-    claim, 
-    key,
-    header = list("kid" = config$boxAppSettings$appAuth$publicKeyID)
-  )
+  # wrap params in function, enable retry with different expiry times
+  params_dt <- function(dt = 30) {
```

Comment:
The naming `dt` doesn't describe the object/idea to me. Would `exp_offset` or `exp_seconds` or something about the fact we are "padding" the system time (to generate the expiration time) be more helpful to a reader?

Reply:
Let me see what I can do here.

Reply:
I changed a few things around and changed the messages.

Thankfully, I was able to "reproduce" the error so I could confirm things work as I expect:

```
Failed JWT request: time offset was 0 seconds.
Retrying JWT request: time offset now -15 seconds.
Failed JWT request: time offset was -15 seconds.
Retrying JWT request: time offset now 15 seconds.
boxr: Authenticated using OAuth2 (JWT) as boxr-jwt-ijl (AutomationUser_868610_9SFx0Bq15D@boxdevedition.com, id: 9862360558)
```

I'll also add some words in the function doc and service-auth vignette - I'll do that in the docs PR